torcnn/static/model_config.txt
coord_conv: False
ps: (128, 256)
img_aug: {}
sample_weights: {}
class_weights: False
batch_norm: True
bias_init: None
scalar_vars: []
channels: ['Reflectivity', 'Velocity', 'RhoHV', 'range_folded_mask', 'out_of_range_mask', 'range']
l2_reg: 0.0
dropout_rate: 0.2
batchsize: 256
num_encoding_blocks: 5
num_decoding_blocks: 0
num_conv_per_block: 2
nfmaps_by_block: [128, 128, 128, 128, 128]
dense_layers: [256, 64]
filter_width: 3
conv_activation: leaky_relu
padding: same
learning_rate: 0.01
nepoch: 100
es_patience: 3
rlr_factor: 0.1
rlr_patience: 1
rlr_cooldown: 2
cnn: cnn
input_tuples: [(128, 256, 6)]
inputs: [['Reflectivity', 'Velocity', 'RhoHV', 'range_folded_mask', 'out_of_range_mask', 'range']]
train_list: ['/work2/jcintineo/torcnn/tfrecs_shard/2011/2011????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2011/2011????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2011/2011????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2011/2011????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2011/2011????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2011/2011????_nonsev*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2012/2012????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2012/2012????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2012/2012????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2012/2012????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2012/2012????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2012/2012????_nonsev*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2013/2013????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2013/2013????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2013/2013????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2013/2013????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2013/2013????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2013/2013????_nonsev*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2014/2014????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2014/2014????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2014/2014????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2014/2014????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2014/2014????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2014/2014????_nonsev*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2015/2015????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2015/2015????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2015/2015????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2015/2015????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2015/2015????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2015/2015????_nonsev*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2016/2016????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2016/2016????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2016/2016????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2016/2016????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2016/2016????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2016/2016????_nonsev*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2017/2017????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2017/2017????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2017/2017????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2017/2017????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2017/2017????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2017/2017????_nonsev*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2018/2018????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2018/2018????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2018/2018????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2018/2018????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2018/2018????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2018/2018????_nonsev*tfrec']
val_list: ['/work2/jcintineo/torcnn/tfrecs_shard/2019/2019????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2019/2019????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2019/2019????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2019/2019????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2019/2019????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2019/2019????_nonsev*tfrec']
targets: ['tornado']
byte_scaling_vals: {'Reflectivity': {'vmin': 5, 'vmax': 70}, 'Velocity': {'vmin': -100, 'vmax': 100}, 'AliasedVelocity': {'vmin': -100, 'vmax': 100}, 'AzShear': {'vmin': -0.01, 'vmax': 0.04}, 'DivShear': {'vmin': -0.01, 'vmax': 0.04}, 'PhiDP': {'vmin': 0, 'vmax': 360}, 'RhoHV': {'vmin': 0, 'vmax': 1}, 'SpectrumWidth': {'vmin': 0, 'vmax': 70}, 'Zdr': {'vmin': -4, 'vmax': 6}, 'azimuth': {'vmin': 0, 'vmax': 360}, 'range': {'vmin': 0, 'vmax': 460}}
binarize: -1
ngpu: 1
loss_fcn: binary_crossentropy
optimizer: AdamW
outdir: /work2/jcintineo/torcnn/tests/2011-19//test04
pool: max
label_smoothing: 0.1
n_tsamples: 3396557
n_vsamples: 830169
train_years: [2011 2012 2013 2014 2015 2016 2017 2018]
val_years: [2019]
pos_classes: ['tor', 'pretor_15', 'pretor_30']
neg_classes: ['hail', 'wind', 'nonsev']


torcnn/static/model/model_config.txt
coord_conv: False
ps: (128, 256)
img_aug: {}
sample_weights: {}
class_weights: False
batch_norm: True
bias_init: None
scalar_vars: []
channels: ['Reflectivity', 'Velocity', 'RhoHV', 'range_folded_mask', 'out_of_range_mask', 'range']
l2_reg: 0.0
dropout_rate: 0.2
batchsize: 256
num_encoding_blocks: 5
num_decoding_blocks: 0
num_conv_per_block: 2
nfmaps_by_block: [128, 128, 128, 128, 128]
dense_layers: [256, 16]
filter_width: 3
conv_activation: leaky_relu
padding: same
learning_rate: 0.01
nepoch: 100
es_patience: 3
rlr_factor: 0.1
rlr_patience: 1
rlr_cooldown: 2
cnn: cnn
input_tuples: [(128, 256, 6)]
inputs: [['Reflectivity', 'Velocity', 'RhoHV', 'range_folded_mask', 'out_of_range_mask', 'range']]
train_list: ['/work2/jcintineo/torcnn/tfrecs_shard/2011/2011????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2011/2011????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2011/2011????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2011/2011????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2011/2011????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2011/2011????_nonsev*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2012/2012????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2012/2012????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2012/2012????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2012/2012????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2012/2012????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2012/2012????_nonsev*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2013/2013????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2013/2013????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2013/2013????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2013/2013????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2013/2013????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2013/2013????_nonsev*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2014/2014????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2014/2014????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2014/2014????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2014/2014????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2014/2014????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2014/2014????_nonsev*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2015/2015????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2015/2015????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2015/2015????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2015/2015????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2015/2015????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2015/2015????_nonsev*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2016/2016????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2016/2016????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2016/2016????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2016/2016????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2016/2016????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2016/2016????_nonsev*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2017/2017????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2017/2017????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2017/2017????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2017/2017????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2017/2017????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2017/2017????_nonsev*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2018/2018????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2018/2018????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2018/2018????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2018/2018????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2018/2018????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2018/2018????_nonsev*tfrec']
val_list: ['/work2/jcintineo/torcnn/tfrecs_shard/2019/2019????_tor*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2019/2019????_pretor_15*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2019/2019????_pretor_30*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2019/2019????_hail*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2019/2019????_wind*tfrec', '/work2/jcintineo/torcnn/tfrecs_shard/2019/2019????_nonsev*tfrec']
targets: ['tornado']
byte_scaling_vals: {'Reflectivity': {'vmin': 5, 'vmax': 70}, 'Velocity': {'vmin': -100, 'vmax': 100}, 'AliasedVelocity': {'vmin': -100, 'vmax': 100}, 'AzShear': {'vmin': -0.01, 'vmax': 0.04}, 'DivShear': {'vmin': -0.01, 'vmax': 0.04}, 'PhiDP': {'vmin': 0, 'vmax': 360}, 'RhoHV': {'vmin': 0, 'vmax': 1}, 'SpectrumWidth': {'vmin': 0, 'vmax': 70}, 'Zdr': {'vmin': -4, 'vmax': 6}, 'azimuth': {'vmin': 0, 'vmax': 360}, 'range': {'vmin': 0, 'vmax': 460}}
binarize: -1
ngpu: 1
loss_fcn: binary_crossentropy
optimizer: AdamW
outdir: /work2/jcintineo/torcnn/tests/2011-19//test05
pool: max
label_smoothing: 0.1
n_tsamples: 3396557
n_vsamples: 830169
train_years: [2011 2012 2013 2014 2015 2016 2017 2018]
val_years: [2019]
pos_classes: ['tor', 'pretor_15', 'pretor_30']
neg_classes: ['hail', 'wind', 'nonsev']


torcnn/detection/config.py
def config():

    c = {}
    c['outdir'] = '/raid/jcintineo/torcnn/detection/tests/test02'
    c['channels'] = ['Reflectivity', 'Velocity', 'RhoHV']
    c['PS'] = (512, 512)
    c['batchsize'] = 64
    c['bsinfo'] = {
        'Reflectivity':{'min':0., 'max':75.0},
        'Velocity':{'min':-80, 'max':80.0},
        'SpectrumWidth':{'min':0, 'max':70.0},
        'RhoHV':{'min':0.4, 'max':1.0},
        'Zdr': {'min':-4., 'max':6.},
    }
    c['obj_weight'] = 1.0  #50.0
    c['noobj_weight'] = 1.0 #0.1
    c['tvs_importance'] = 5.0
    c['lr'] = 5e-5
    c['rlr_cooldown'] = 2
    c['rlr_patience'] = 1
    c['rlr_factor'] = 0.1
    c['es_patience'] = 3
    c['monitoring_thresholds'] = [0.1, 0.3, 0.5, 0.7]   
 
    tfrec_dir = "/raid/jcintineo/torcnn/detection/tfrecs_100km60min"
    # N.B. Can't have any "//" in the train_list or val_list!!!
    c['train_list'] = [f"{tfrec_dir}/201?/20*/*tfrec", f"{tfrec_dir}/202[0-2]/20*/*tfrec"] 
    c['val_list'] = [f"{tfrec_dir}/2023/20*/*tfrec"]
       
    return c
